functional_tests/steps/footer_menu.py

```diff
@@ -0,0 +1,225 @@
+    fill_column_link(
+        context.page, col_index=0, link_index=0, external_url="htp:/malformed-url", link_title="Malformed URL"
+    )
+
+
+@then("an error message is displayed about the URL format")
+def url_format_error(context: Context):
+    expect(context.page.locator("#columns-0-value-links-0-value-external_url-errors")).to_contain_text(
+        "Enter a valid URL."
+    )
+
+
+@when("the user adds more than 3 columns")
+def user_inserts_more_than_max_columns(context: Context):
+    generate_columns(context, num_columns=4)
+
+
+@then("an error message is displayed about column limit")
+def max_column_error(context: Context):
+    expect(context.page.locator("#panel-columns-content")).to_contain_text("The maximum number of items is 3")
+
+
+@when("the user adds more than 10 links")
+def user_adds_above_max_links(context: Context):
+    insert_block(context.page)
+    fill_column_title(context.page, col_index=0, title="Link Column 1")
+
+    for i in range(11):
+        fill_column_link(
+            context.page,
+            col_index=0,
+            link_index=i,
+            external_url=f"https://www.link-column-1-link-{i}.com/",
+            link_title=f"Link Title {i + 1}",
+        )
+
+
+@then("an error message is displayed about the link limit")
+def max_link_error(context: Context):
+    expect(context.page.locator("#panel-columns-content")).to_contain_text("The maximum number of items is 10")
+
+
+@when("the user navigates to Snippets")
+def user_navigates_to_snippets(context: Context):
+    context.page.get_by_role("link", name="Snippets").click()
+
+
+@when('the user clicks on "Footer menus"')
+def user_clicks_footer_menus(context: Context):
+    context.page.get_by_role("link", name="Footer menus").click()
+
+
+@when('the user selects "More options for Footer Menu"')
+def user_selects_more_options(context: Context):
+    context.page.get_by_role("button", name="More options for 'Footer Menu'").click()
+
+
+@when('the user clicks "Delete Footer Menu"')
+def user_selects_delete_footer_menu(context: Context):
+    context.page.get_by_role("link", name="Delete 'Footer Menu'").click()
+    context.page.get_by_role("button", name="Yes, delete").click()
+
+
+@then("a banner confirming the deletion is displayed")
+def user_sees_deletion_confirmation(context: Context):
+    expect_alert_banner(context.page, "Footer menu 'Footer Menu' deleted.")
+
+
+@when("a footer menu is populated with 3 columns")
+def create_populated_footer_menu(context: Context):
+    generate_columns(context, num_columns=3)
+
+
+@then("the footer menu appears on the home page")
+def footer_menu_appears_on_home_page(context: Context):
+    # Check that the footer menu appears on the home page
+    # Extract the current port from the page's URL
+    current_url = context.page.url
+    port = current_url.split(":")[2].split("/")[0]
+
+    # Navigate to the desired URL with the extracted port
+    context.page.goto(f"http://localhost:{port}/")
```

Comment:
This isn't necessary, `context` has the `base_url` which you can navigate to:

```suggestion
    context.page.goto(context.base_url)
```

Reply:
Done [7f23b29](https://github.com/ONSdigital/dis-wagtail/pull/130/commits/7f23b29a24646eef450fe1de8e0d50b906b29a50)